src/test/ui/assign-to-method.rs

```diff
@@ -1,3 +1,5 @@
+// compile-flags: -Zsave-analysis
```

Comment:
I think we should have a dedicated regression test in `src/test/ui/save-analysis` and not rely on this flag in general `src/test/ui` directory

cc @Centril on test layout

Reply:
My general sentiment is that we should not have files at all in `src/test/ui`, only directories. I have a question though... is the purpose of this test just for save-analysis? If not, then it doesn't seem fit for a `save-analysis/` folder, which I would expect to be targeted specifically at testing save-analysis as opposed to some other aspect (and save analysis being incidentally also tested).

Reply:
Left comments to clarify and moved `assign-to-method.rs` to `ui/methods`.